### ISSUE TITLE
`manual_assert` does not simplify expressions

### ISSUE THREAD

**Issue by Velocifyer** [rust-lang/rust-clippy#15359]
### Summary

manual_assert does not always simplify expressions when generating a suggestion on how to fix it causing `nonminimal_bool` to happen when applying suggestions

### Reproducer

I tried this code:

```rust
#![warn(clippy::manual_assert)]
pub fn foo(bar: u128) {
    if bar != 0 {
        panic!("How fun it is too play slither io wow it is so fun go go go")
    }
}
```

I expected to see this happen:
Clippy suggests `assert!(bar == 0, "How fun it is too play slither io wow it is so fun go go go);` 

Instead, this happened:
clippy suggests`assert!(!(bar != 0), "How fun it is too play slither io wow it is so fun go go go");`
witch causes 
```
warning: this boolean expression can be simplified
 --> src/lib.rs:3:13
  |
3 |     assert!(!(bar != 0), "How fun it is too play slither io wow it is so fun go go go");
  |             ^^^^^^^^^^^ help: try: `(bar == 0)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
  = note: `#[warn(clippy::nonminimal_bool)]` on by default
```

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: x86_64-unknown-linux-gnu
release: 1.88.0
LLVM version: 20.1.5
```

**Comment by Velocifyer** [comment]
This might be a feature request and not a bug

**Comment by Velocifyer** [comment]
This also happens with `>=`
